### PR TITLE
DetectionSensor: more flexible triggering

### PR DESCRIPTION
**Pull request by augustozanellato**
This PR introduces support for more triggering types in detection sensor, specifically:
 - `LOGIC_LOW`: same as previous behavior with `detection_triggered_high` set to false: a "detection message" is sent every `minimum_broadcast_secs` if pin is low;
 - `LOGIC_HIGH`: same as previous behavior with `detection_triggered_high` set to true: a "detection message" is sent every `minimum_broadcast_secs` if pin is high;
 - `FALLING_EDGE`: a single "detection message" is sent when pin goes from high to low;
 - `RISING_EDGE`: a single "detection message" is sent when pin goes from low to high;
 - `EITHER_EDGE_ACTIVE_LOW`: a single "detection message" is sent when pin goes from high to low, another single "state message" is sent when pin goes from low to high;
 - `EITHER_EDGE_ACTIVE_HIGH`: a single "detection message" is sent when pin goes from low to high, another single "state message" is sent when pin goes from high to low;

Fixes https://github.com/meshtastic/firmware/issues/4753.

Depends on meshtastic/protobufs#582 to be merged first.

### CODE REVIEW

File: src/modules/DetectionSensorModule.cpp [hunk 1]
```diff
@@ -50,20 +86,29 @@ int32_t DetectionSensorModule::runOnce()
 
     // LOG_DEBUG("Detection Sensor Module: Current pin state: %i\n", digitalRead(moduleConfig.detection_sensor.monitor_pin));
 
-    if (!Throttle::isWithinTimespanMs(lastSentToMesh,
-                                      Default::getConfiguredOrDefaultMs(moduleConfig.detection_sensor.minimum_broadcast_secs)) &&
-        hasDetectionEvent()) {
-        sendDetectionMessage();
-        return DELAYED_INTERVAL;
+    if ((millis() - lastSentToMesh) >= Default::getConfiguredOrDefaultMs(moduleConfig.detection_sensor.minimum_broadcast_secs)) {
```
Comment (thebentern) [inline src/modules/DetectionSensorModule.cpp:89]: Use the new:
```
!Throttle::isWithinTimespanMs(lastSentToMesh,
                                      Default::getConfiguredOrDefaultMs(moduleConfig.detection_sensor.minimum_broadcast_secs)
```

File: src/modules/DetectionSensorModule.cpp [hunk 2]
```diff
@@ -50,20 +86,29 @@ int32_t DetectionSensorModule::runOnce()
...
     // Even if we haven't detected an event, broadcast our current state to the mesh on the scheduled interval as a sort
     // of heartbeat. We only do this if the minimum broadcast interval is greater than zero, otherwise we'll only broadcast state
     // change detections.
-    else if (moduleConfig.detection_sensor.state_broadcast_secs > 0 &&
-             !Throttle::isWithinTimespanMs(lastSentToMesh,
-                                           Default::getConfiguredOrDefaultMs(moduleConfig.detection_sensor.state_broadcast_secs,
-                                                                             default_telemetry_broadcast_interval_secs))) {
-        sendCurrentStateMessage();
+    if (moduleConfig.detection_sensor.state_broadcast_secs > 0 &&
+        (millis() - lastSentToMesh) >= Default::getConfiguredOrDefaultMs(moduleConfig.detection_sensor.state_broadcast_secs,
```
Comment (thebentern) [inline src/modules/DetectionSensorModule.cpp:109]: Use the new 
```
!Throttle::isWithinTimespanMs(lastSentToMesh,
                                           Default::getConfiguredOrDefaultMs(moduleConfig.detection_sensor.state_broadcast_secs,
                                                                             default_telemetry_broadcast_interval_secs)
```

**Comment by thebentern** [comment]
@augustozanellato looks great! I added some feedback about using the new Throttle utility method for time windowing. Once those changes are in, I think we're good to merge.

**Comment by augustozanellato** [comment]
Changes addressed!
Btw trunk is failing on a file I haven't touched so probably it's borked on master:
```
.github/workflows/tests.yml:60:14
  -:-   fmt   Incorrect formatting, autoformat by running 'trunk fmt'                                                                                                                            prettier               
 60:14  high  label "test-runner" is unknown. available labels are "windows-latest", "windows-latest-8-cores", "windows-2022", "windows-2019", "ubuntu-latest", "ubuntu-latest-4...              actionlint/runner-label
              .trunk/out/B7X.txt 
```

**Comment by augustozanellato** [comment]
I missed a `)`, now everything should build fine

**Comment by augustozanellato** [comment]
Failed jobs seem to be caused by ci flakyness (and probably issues on github's end), a restart should fix them

**Review by thebentern — APPROVED** [review]

**Review by caveman99 — APPROVED** [review]

**Pull request merged**